packages/cli/src/PublicApi/v1/handlers/tags/tags.handler.ts

```diff
@@ -0,0 +1,143 @@
+import type express from 'express';
+
+import { getTags, getTagsCount, createTag, updateTag, deleteTag, getTagById } from './tags.service';
+import { TagEntity } from '@db/entities/TagEntity';
+import { authorize, validCursor } from '../../shared/middlewares/global.middleware';
+import type { TagRequest } from '../../../types';
+import { encodeNextCursor } from '../../shared/services/pagination.service';
+import { ExternalHooks } from '@/ExternalHooks';
+import { validateEntity } from '@/GenericHelpers';
+
+import { Container } from 'typedi';
+import type { FindManyOptions, FindOptionsWhere } from '@n8n/typeorm';
+
+export = {
+	createTag: [
+		authorize(['global:owner', 'global:admin', 'global:member']),
+		async (req: TagRequest.Create, res: express.Response): Promise<express.Response> => {
+			const { name } = req.body;
+
+			const newTag = new TagEntity();
+			newTag.name = name.trim();
+
+			await Container.get(ExternalHooks).run('tag.beforeCreate', [newTag]);
+
+			await validateEntity(newTag);
+
+			const where: FindOptionsWhere<TagEntity> = {
+				...(newTag.name !== undefined && { name: newTag.name }),
```

Comment:
Hi @ivov , I don't think it can be undefined. There is a test that sends an empty payload (`{}`), and you get the expected 400 response. The same thing if the `name` property in the JSON is an empty string or a `null`.

Reply:
Hi @ivov, I have removed the condition `newTag.name !== undefined` in #8595